### PR TITLE
[ODS-4903] Load tool - year specific mode issue

### PR DESCRIPTION
**Pull request by jamessmckay**

### CODE REVIEW

File: Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfiguration.cs [hunk 1]
```diff
@@ -171,7 +171,7 @@ string ResolvedUrl(string url)
             {
                 return apiMode == ApiMode.YearSpecific
                     ? Regex.Replace(
-                        url, @"(?<year>\d{4})\b", configuration.GetValue<string>("OdsApi:SchoolYear"), RegexOptions.None)
+                        url, @"\/(?<year>\b\d{4}\b)\/", configuration.GetValue<string>("OdsApi:SchoolYear"), RegexOptions.None)
```
Comment (blmeyers) [inline Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfiguration.cs:174]: This applies to all the regex usage.  You're requiring a trailing slash now on this url, and the api endpoint doesn't actually generate one, so this regex doesn't work to match for several of the urls.

This is also still aggressively catching 4 digits earlier in the url, but it is definitely a lot less aggressive than it was. If we can get it handling without the trailing slash that's probably fine for now, but eventually we'd probably want to handle this by only replacing the last match and not all matches.

See https://api-stage.ed-fi.org/YearSpecific_v5.2/api/.

**Review by blmeyers — CHANGES_REQUESTED** [review]

File: Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfiguration.cs [hunk 2]
```diff
@@ -171,7 +171,9 @@ string ResolvedUrl(string url)
             {
                 return apiMode == ApiMode.YearSpecific
                     ? Regex.Replace(
-                        url, @"(?<year>\d{4})\b", configuration.GetValue<string>("OdsApi:SchoolYear"), RegexOptions.None)
+
+                        // https://regex101.com/r/lgugB1/1
```
Comment (blmeyers) [inline Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfiguration.cs:175]: Can you update this to https://regex101.com/r/KywmUK/1 everywhere to reflect your changes to the Regex replace.

**Review by blmeyers — CHANGES_REQUESTED** [review]
Minor documentation update, but seems to be working with local testing.

File: Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfiguration.cs (hunk 2 above)
Comment (tbmeador-dlp) [reply Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfiguration.cs]: done

**Review by blmeyers — APPROVED** [review]

**Pull request merged**